arch/cortex-m0/src/lib.rs

```diff
@@ -77,7 +73,7 @@ pub unsafe extern "C" fn switch_to_user(mut user_stack: *const u8,
 
     "
     : "=r"(user_stack)
-    : "r"(user_stack), "r"(process_got), "r"(process_regs)
+    : "r"(user_stack), "r"(process_regs)
```

Comment:
Basically, if we add a clobber for `r0`, I'm not sure I understand why we wouldn't add a clobber for _all_ registers (basically that leaves `r1-r3`), in which case the compiler will complain that it can't find any registers to allocate.